### PR TITLE
i18n(de): Add `core-concepts/partial-hydration.md`

### PR DESCRIPTION
**Pull request by hippotastic**
#### What kind of changes does this PR include?

- [ ] Minor content fixes (broken links, typos, etc.)
- [ ] New or updated content
- [X] Translated content
- [ ] Changes to the docs site code
- [ ] Something else!

#### Description

- Adds German translation for `core-concepts/partial-hydration.md`.
- Note: Since the diagram linked in the original article is only available in English, I have created a component that recreates the diagram using grid layout. This allows easy translation and makes it accessible.

### CODE REVIEW

**Comment by hippotastic** [comment]
Regarding the failed check: The links are fine. The build process just had a random error.

File: src/components/IslandsDiagram.astro [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		align-items: center;
+		text-align: center;
+		padding: 2rem;
+		background: hsl(0, 0%, 96%);
+		border: 2px solid hsl(0, 0%, 80%);
+		color: hsl(0, 0%, 40%);
+	}
+	.diagram .app {
+		font-size: 1.05rem;
+	}
```
Comment (delucis) [inline src/components/IslandsDiagram.astro:41]: I’m not sure this is doing much? I’d either bump up the font size a little more (`1.25rem` looks more like the original image?) or just remove this and have them all be the same size.

Comment (delucis) [reply src/components/IslandsDiagram.astro:41]: Yeah, I noticed that too. OK we’ll roll with it!

**Review by delucis — APPROVED** [review]
Looks good on a quick look and merging to unblock usage of the IslandsDiagram component in other languages. Thanks @hippotastic!

**Pull request merged**